dotCMS/src/main/java/com/dotcms/rest/api/v1/taillog/TailLogResource.java

```diff
@@ -0,0 +1,235 @@
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import org.glassfish.jersey.media.sse.EventOutput;
+import org.glassfish.jersey.media.sse.OutboundEvent;
+import org.glassfish.jersey.media.sse.SseFeature;
+import org.glassfish.jersey.server.JSONP;
+
+/**
+ * This resource provides the endpoint used by the LogViewer functionality to display backend server logs
+ * @author nollymarlonga
+ */
+@Path("/v1/tailLog")
+@Tag(name = "TailLog")
+public class TailLogResource {
+
+    private static final int LINES_PER_PAGE = 10;
+
+
+    @GET
+    @Path("/{fileName}")
+    @JSONP
+    @NoCache
+    @Produces(SseFeature.SERVER_SENT_EVENTS)
+    public final EventOutput getLogs(@Context final HttpServletRequest request,
+            @PathParam("fileName") final String fileName, @PathParam("linesBack") final int linesBack) throws IOException {
+
+        new WebResource().init(null, request, new EmptyHttpResponse(), true, null);
+
+
+        if(fileName.trim().isEmpty()) {
+            return sendError("File should not be empty");
+        }
+
+
+        final String sanitizedFileName = FileUtil.sanitizeFileName(fileName);
+        String tailLogLofFolder = Config.getStringProperty("TAIL_LOG_LOG_FOLDER", "./dotsecure/logs/");
+        if (!tailLogLofFolder.endsWith(File.separator)) {
+            tailLogLofFolder = tailLogLofFolder + File.separator;
+        }
+
+        final File logFolder 	= new File(FileUtil.getAbsolutlePath(tailLogLofFolder));
+        final File logFile 	= new File(FileUtil.getAbsolutlePath(tailLogLofFolder + sanitizedFileName));
+
+
+        // if the logFile is outside of the logFolder, die
+        if ( !logFolder.exists() || !logFolder.canRead()
+                ||   !logFile.getCanonicalPath().startsWith(logFolder.getCanonicalPath())) {
+
+            final String errorMessage = "Invalid File request:" + logFile.getCanonicalPath() + " from:" +request.getRemoteHost();
+            SecurityLogger.logInfo(
+                    TailLogResource.class,  errorMessage);
+            return sendError(errorMessage);
+        }
+
+        Logger.info(this.getClass(), "Requested logFile:" + logFile.getCanonicalPath());
+
+        final String regex = Config.getStringProperty("TAIL_LOG_FILE_REGEX", ".*\\.log$|.*\\.out$");
+
+
+        if(!Pattern.compile(regex).matcher(sanitizedFileName).matches()){
+            sendError("Error ");
+        }
+
+        final MyTailerListener listener = new MyTailerListener();
+
+
+        final MyTailerThread thread = new MyTailerThread(logFile, listener, new EventOutput(), linesBack);
+
+        String name = null;
+
+        //Finds a new thread to read the log
+        for (int i = 0; i < 1000; i++) {
+            name = "LogTailer" + i + ":" + sanitizedFileName;
+            Thread t = ThreadUtils.getThread(name);
+            if (t == null) {
+                break;
+            }
+            if (i > 100) {
+                sendError("Too many Logger threads");
```

Comment:
same here

Reply:
It writes out an event of type `org.glassfish.jersey.media.sse.EventOutput`
it is not supposed to carry an HTTP Code